frontend/src/TestEdit.js

```diff
@@ -0,0 +1,267 @@
+        {
+            onSuccess: (data)=> correctItem(data),
+            enabled: id !== 'new'
+        }
+    );
+
+    if (error) return <div>Request Failed</div>;
+    if (isLoading) return <div>Loading...</div>;
+
+
+    function handleGroupChange(event) {
+        const target = event.target;
+        const value = target.value;
+        let newItem = {...item};
+
+
+        if(/(([a-z]{1,})|([A-Z]{1,}))-[1-9]{2}/.test(value)) {
+            fetch('/subload/check_group/'+value)
+                .then(response => {
+                    if (response.ok) {
+                        return response.json();
+                    } else {
+                        throw new Error('Failed to fetch boolean value');
+                    }
+                })
+                .then(data => {
+                    if (data) {
+                        target.style.backgroundColor = "green";
+                    } else {
+                        target.style.backgroundColor = "red";}
+                })
+                .catch(error => {
+                    console.error(error);
+                });
+
+            }else{
+                target.style.backgroundColor = "red";
+            }
+
+        newItem.groupName = value;
+        setItem(newItem);
+    }
+
+
+    function handleChange(event) {
+        const target = event.target;
+        const value = target.value;
+        const name = target.name;
+        let newItem = {...item};
+
+        newItem.test[name] = value;
+        setItem(newItem);
+    }
+
+
+    function handleCheckboxChange(id, sId, event) {
+        let newItem = {...item};
+        const target = event.target;
+
+        let q =newItem.questions.find(q => q.id === +id);
+        if(!target.checked){
+            (q.answers) = (q.answers).filter(item => item !== +sId );
+        }else {
+            if(!q.answers.includes(+sId)){
+                (q.answers).push(+sId);
+            }
+        }
+            setItem(newItem);
+        }
+
+    function handleQuestionsChange( id,  sId =null,event ) {
+        const value =  event.target.value;
+        let newItem = {...item};
+
+        if(sId===null){
+            newItem.questions[id].question = value;
+        }else {
+            (newItem.questions[id]).answerVariants[sId] = value;
+        }
+        setItem(newItem);
+    }
+
+
+    async function handleSubmit(event) {
+        event.preventDefault();
+        item.added= Array.from(item.added);
+        item.deleted= Array.from(item.deleted);
+
+        await fetch('/subload/tests' + (item.test.id ? '/' + item.test.id : ''), {
+            method: (item.test.id) ? 'PUT' : 'POST',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(item),
+        });
+    }
+
+
+    async function delAnswer(index, sIndex) {
+        let newItem = {...item};
+        let q =newItem.questions[index]
```

Comment:
fix spacing, ensure there is always one space before and after =

Reply:
Done